src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -209,6 +209,10 @@ def restore_training_state(self) -> None:
     def restore_precision_plugin_state(self) -> None:
         """Restore the precision plugin state from the pre-loaded checkpoint."""
         prec_plugin = self.trainer.precision_plugin
+
+        if isinstance(prec_plugin, ApexMixedPrecisionPlugin) and self.trainer.state.fn == TrainerFn.FITTING:
```

Comment:
What about a `Trainer` without apex enabled loaded with a checkpoint that has apex state? That doesn't raise. Should it?